tests/test_common.py

```diff
@@ -113,90 +125,61 @@ def test_copy(grid):
     assert np.all(grid_copy_shallow.points[0] == grid.points[0])
 
 
-def test_transform(grid):
-    trans = vtk.vtkTransform()
-    trans.RotateX(30)
-    trans.RotateY(30)
-    trans.RotateZ(30)
-    trans.Translate(1, 1, 2)
-    trans.Update()
-
-    grid_a = grid.copy()
-    grid_b = grid.copy()
-    grid_c = grid.copy()
-    grid_a.transform(trans)
-    grid_b.transform(trans.GetMatrix())
-    grid_c.transform(pyvista.trans_from_matrix(trans.GetMatrix()))
-    assert np.allclose(grid_a.points, grid_b.points)
-    assert np.allclose(grid_a.points, grid_c.points)
-
-
-def test_transform_errors(grid):
-    with pytest.raises(TypeError):
-        grid.transform(None)
-
-    with pytest.raises(Exception):
-        grid.transform(np.array([1]))
-
-
-def test_translate(grid):
-    grid_copy = grid.copy()
-    xyz = [1, 1, 1]
-    grid_copy.translate(xyz)
-
-    grid_points = grid.points.copy() + np.array(xyz)
-    assert np.allclose(grid_copy.points, grid_points)
-
-
-def test_rotate_x(grid):
-    angle = 30
-    trans = vtk.vtkTransform()
-    trans.RotateX(angle)
-    trans.Update()
-
-    trans_filter = vtk.vtkTransformFilter()
-    trans_filter.SetTransform(trans)
-    trans_filter.SetInputData(grid)
-    trans_filter.Update()
-    grid_a = pyvista.UnstructuredGrid(trans_filter.GetOutput())
-
-    grid_b = grid.copy()
-    grid_b.rotate_x(angle)
-    assert np.allclose(grid_a.points, grid_b.points)
-
-
-def test_rotate_y(grid):
-    angle = 30
-    trans = vtk.vtkTransform()
-    trans.RotateY(angle)
-    trans.Update()
-
-    trans_filter = vtk.vtkTransformFilter()
-    trans_filter.SetTransform(trans)
-    trans_filter.SetInputData(grid)
-    trans_filter.Update()
-    grid_a = pyvista.UnstructuredGrid(trans_filter.GetOutput())
-
-    grid_b = grid.copy()
-    grid_b.rotate_y(angle)
-    assert np.allclose(grid_a.points, grid_b.points)
-
-
-def test_rotate_z(grid):
-    angle = 30
-    trans = vtk.vtkTransform()
-    trans.RotateZ(angle)
-    trans.Update()
-
-    trans_filter = vtk.vtkTransformFilter()
-    trans_filter.SetTransform(trans)
-    trans_filter.SetInputData(grid)
-    trans_filter.Update()
-    grid_a = pyvista.UnstructuredGrid(trans_filter.GetOutput())
-
-    grid_b = grid.copy()
-    grid_b.rotate_z(angle)
-    assert np.allclose(grid_a.points, grid_b.points)
+class TestTransform:
```

Comment:
Purely personal preference (i.e. feel free to ignore):  
I prefer flat over nested whenever possible. In this case I'd prefer grouping in files or no grouping at all over classes (assuming the class has no other purpose than grouping).

Reply:
I agree with this.  Classes also make it a bit harder to debug as well.  We can still group this using:
```python
###############################################################################
# Transform
###############################################################################
```

Reply:
More specifically the error is not with vtk but with numpy:

    TypeError: ufunc 'add' output (typecode 'O') could not be coerced to provided output parameter (typecode 'd') according to the casting rule ''same_kind''

Reply:
And on classes, it makes sense to me to have not only grouping by file, but by the specific method being tested. Why specifically would it be harder to debug if they're grouped in this way? I can see it could be harder to read on a console/terminal, pycharm gives a nice tree structure when grouped like this so I'm probably biased. But if multiple people agree it's easier to consider not doing so.

Reply:
I almost feel somewhat bad having this brought up, this should definitely not block this change. It's purely style and while consistent coding style is good, there's neither a precedent for this in pyvista nor frankly is styling in general very consistent (or at least not enforced to be). And I don't get the difficulty regarding debugging either.

Reply:
The debugging behavior is a bit out of laziness on my end.  I'm fine with grouping by classes, and I agree that it makes the code easier to follow and to know where to add additional tests.

Reply:
FWIW, I don't like grouping by classes in tests - it's unnecessary.

Reply:
@imsodin 

No worries, it's a big style change so it's good to have input/consensus on this. I'm just not sure how I want to handle the other issues.

Reply:
Other issues can be fixed by not mitigating the failures that occur on VTK's side just yet, though it's helpful to know that they occur and we might consider mitigating it in the future on our side.  I agree with @imsodin about garbage in/garbage out and we should implement `equal_nan=True`.